component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateUtil.java

```diff
@@ -380,22 +380,29 @@ private static boolean isUserExists(String userName, AuthenticationContext authe
     }
 
     /**
-     * Check whether user account is locked or not.
+     * Get the user account state by checking  whether the user account is locked or not.
      *
-     * @param user Authenticated user.
+     * @param subject userName that passed from the cert CN value.
      * @return boolean account locked or not.
      * @throws AccountLockServiceException
      */
-    public static boolean isAccountLock(AuthenticatedUser user) throws AccountLockServiceException {
+    public static boolean isAccountLock(String subject) throws AccountLockServiceException {
```

Comment:
Create a new method as below.
```suggestion
    public static boolean isAccountLock(String subject) throws AccountLockServiceException {
```